apps/dashboard/src/components/workflow-editor/channel-preferences-form.tsx

```diff
@@ -0,0 +1,305 @@
+          </Link>
+        </SidebarHeader>
+        <Separator />
+
+        {isDashboardWorkflow ? null : (
+          <>
+            <SidebarContent size="md">
+              <p className="text-xs text-neutral-400">
+                Set default notification channels for subscribers and specify which channels they can customize.
+              </p>
+              {/* This doesn't needs to be a form, but using it as a form allows to re-use the formItem designs without duplicating the same styles */}
+              <Form {...overrideForm}>
+                <form>
+                  <FormField
+                    control={overrideForm.control}
+                    name="override"
+                    render={({ field }) => (
+                      <FormItem className="mt-2 flex w-full items-center justify-between">
+                        <FormLabel tooltip="Override preferences to use dashboard-defined preferences instead of code defaults. Disable to restore defaults.">
+                          Override preferences
+                        </FormLabel>
+                        <FormControl>
+                          <Switch
+                            checked={field.value}
+                            onCheckedChange={(checked) => {
+                              field.onChange(checked);
+                              if (!checked) {
+                                updateUserPreference(null);
+                              }
+                              track(TelemetryEvent.WORKFLOW_PREFERENCES_OVERRIDE_USED, {
+                                new_status: checked,
+                              });
+                            }}
+                            disabled={isDashboardWorkflow}
+                          />
+                        </FormControl>
+                      </FormItem>
+                    )}
+                  />
+                </form>
+              </Form>
+            </SidebarContent>
+            <Separator />
+          </>
+        )}
+        <Form {...form}>
+          <form>
+            <SidebarContent size="md">
+              <FormField
+                control={form.control}
+                name="user.all.readOnly"
+                render={({ field }) => (
+                  <FormItem className="my-2 flex w-full items-center justify-between">
+                    <FormLabel tooltip="Critical workflows ensure essential notifications can't be unsubscribed.">
+                      Mark as critical
+                    </FormLabel>
+                    <FormControl>
+                      <Switch checked={field.value} onCheckedChange={handleCriticalToggle} disabled={!override} />
+                    </FormControl>
+                  </FormItem>
+                )}
+              />
+            </SidebarContent>
+            <div className="flex items-center justify-between gap-1.5 bg-neutral-50 px-3 py-0.5">
+              <span className="text-2xs uppercase text-neutral-400">channels</span>
```

Comment:
For dashboard-defined workflows, this checkbox shouldn't be visible. For code-first, the copywriting should be "Override code defaults"